packages/jsii/test/deprecation-warnings.test.ts

```diff
@@ -503,7 +851,54 @@ function jsFile(result: HelperCompilationResult, baseName = 'index'): string {
     ([name]) => name === `${baseName}.js`,
   );
 
-  return file![1];
+  if (!file) {
+    throw new Error(`Could not find file with base name: ${baseName}`);
+  }
+
+  return file[1];
+}
+
+function createVmContext(compilation: HelperCompilationResult) {
+  const context = vm.createContext({
+    exports: {},
+    process: {
+      env: {
+        JSII_DEPRECATED: 'fail',
+      },
+    },
+    // Bringing in a "fake" require(id) function that'll resolve relative paths
+    // to files within the compilation output, and module names using the
+    // regular require. When loading a file that was part of the compiler output,
+    // this emulates the situation that would be if the file had been through a
+    // bundler by turning all sequences of white spaces (new line included) into
+    // single spaces.
+    require: (id: string) => {
```

Comment:
Is there a better way?

Reply:
That's the sort of dance you have to do when you use VM contexts in `node`. The context keys are global values that'll be available within the sandbox; which otherwise only has access to ES primordials... So you need to give it "a" require function, and that function is a pretty standard implementation here...

All those APIs are stable enough that they won't break under us when we upgrade to a new node version, too.